src/util/ui.hpp

```diff
@@ -36,4 +36,18 @@ namespace util::ui {
     PopupLayout getPopupLayout(cocos2d::CCSize popupSize);
 
     cocos2d::CCNode* findChildByMenuSelectorRecursive(cocos2d::CCNode* node, uintptr_t function);
+
+    struct BadgeLayout {
+        std::string badgePNG;
+        float scale;
+        cocos2d::CCPoint pos;
+        std::string id;
+        cocos2d::CCNode* badgeWrapper;
+    };
+
+    BadgeLayout createLayout(std::string _badgePNG, float _scale, cocos2d::CCPoint _pos, std::string _id, cocos2d::CCNode* _badgeWrapper);
+
+    uibuilder::Build<cocos2d::CCSprite> createBadge(BadgeLayout layout);
+
+    uibuilder::Build<cocos2d::CCSprite> createBadgeIfSpecial(cocos2d::ccColor3B color, BadgeLayout layout);
```

Comment:
i don't see the necessity of all the options here. imo position, scale, and adding the node to another node should be managed by the caller, instead of these functions. all it should do is just accept a `ccColor3B`, and create a ccsprite with the corresponding badge image. optionally i guess setting the node ID isn't a bad idea, but the rest feels unnecessary.